x/foundation/keeper/abci.go

```diff
@@ -0,0 +1,24 @@
+package keeper
+
+import (
+	"time"
+
+	"github.com/line/lbm-sdk/telemetry"
+	sdk "github.com/line/lbm-sdk/types"
+	"github.com/line/lbm-sdk/x/foundation"
+)
+
+// BeginBlocker withdraws rewards from fee-collector before the distribution
+// module's withdraw.
+func BeginBlocker(ctx sdk.Context, k Keeper) {
+	defer telemetry.ModuleMeasureSince(foundation.ModuleName, time.Now(), telemetry.MetricKeyBeginBlocker)
+
+	if err := k.CollectFoundationTax(ctx); err != nil {
+		panic(err)
+	}
```

Comment:
We discussed about this reward ratio, and I have known the treasury reward also distributed the `distribution` module for equaling between the each ratio and genesis ratio. Isn't it? But this code it is not. 

This rule was changed?

Reply:
The planning team decided to keep the current design.
If the decision would be changed, I will introduce another wrapper module for x/distribution. In that case, changing the tax rate would be a parameter of x/distribution, which means we need to get a confirm from x/gov to change the value.

Reply:
The consensus of validators is also required to modify the value of `FoundationTax`. So it is equals. 
But It seems that it is not fair in the blockchain system to first deduct the total reward value and then divide the remainder between the validator and the proposer.

Reply:
I'm not against to modifying the value by x/gov proposal.
I stated the plan because the cost required to redesign foundation tax rate is rather high. And that was the main reason for the planning team to decide to keep the current design.

Reply:
If then, how about adding this as another issue and implementing it next time?